src/control/Control.Layers.js

```diff
@@ -12,9 +12,14 @@ L.Control.Layers = L.Control.extend({
 	initialize: function (baseLayers, overlays, options) {
 		L.setOptions(this, options);
 
+		if (this.options.autoCollapse && !this.options.autoCollapseTimeout) {

```

Comment:
I'll have a look at this and revise it tonight. I agree that simplifying the options would be a good solution.